dialvia/http.go

```diff
@@ -19,9 +19,10 @@ import (
 )
 
 type HTTPProxyDialer struct {
-	dial      ContextDialerFunc
-	proxyURL  *url.URL
-	tlsConfig *tls.Config
+	dial                   ContextDialerFunc
+	proxyURL               *url.URL
+	tlsConfig              *tls.Config
+	ConnectRequestModifier func(req *http.Request) error
```

Comment:
I guess it would be easier if it could not return error.

Think [Rewrite](https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/net/http/httputil/reverseproxy.go;l=125).

Please add to separate group

Reply:
After seeing it wider context it makes sense.

Reply:
I moved it to a separate group.